### PR TITLE
Use from_variable factory method to instantiate QuantizedVariables

### PR DESCRIPTION
**Pull request by lgeiger**
This allows creation of `QuantizedVariable` and `QuantizedDistributedVariable` using `QuantizedVariable.from_variable` following the naming conventions from `tf.data.Dataset.from_{tensors, tensor_slices, generator}`.

Closes #358

### CODE REVIEW

File: larq/quantized_variable.py [hunk 1]
```diff
@@ -39,6 +39,63 @@ def __init__(self, variable, quantizer=None, precision=None):
         self.quantizer = quantizer
         self.precision = precision or getattr(quantizer, "precision", None)
 
+    @classmethod
```
Comment (lgeiger) [inline larq/quantized_variable.py]: We could also use a `staticmethod` here, not sure what's better.

File: larq/layers_base.py [hunk 2]
```diff
@@ -3,7 +3,8 @@
 import tensorflow as tf
 
 from larq import metrics as lq_metrics
-from larq import quantized_scope, quantized_variable, quantizers
+from larq import quantized_scope, quantizers
```
Comment (leonoverweel) [inline larq/layers_base.py]: I think it'd be cleaner to combine these `from larq` imports:

```python
from larq import quantized_scope, quantizers, metrics as lq_metrics
```

File: larq/quantized_variable.py (hunk 1 above)
Comment (leonoverweel) [reply larq/quantized_variable.py]: The [`tf.data.Dataset.from_tensor`](https://github.com/tensorflow/tensorflow/blob/r2.0/tensorflow/python/data/ops/dataset_ops.py#L397) and [`tf.data.Dataset.from_tensor_slices`](https://github.com/tensorflow/tensorflow/blob/r2.0/tensorflow/python/data/ops/dataset_ops.py#L417) factories both use `staticmethod`, but reading [this StackOverflow answer](https://stackoverflow.com/a/15031999/1814949) and [this Software Engineering StackExchange answer](https://softwareengineering.stackexchange.com/a/166715), I think that you're right in using a `classmethod`. Maybe we should actually PR those TF Datasets methods to turn them into `classmethod`. 😄

File: larq/layers_base.py (hunk 2 above)
Comment (lgeiger) [reply larq/layers_base.py]: I agree, but `isort` doesn't 😉

Comment (lgeiger) [reply larq/layers_base.py]: Changed the config in 9381fe27fbfb9ba79b3c0e8187401513944ef695

File: larq/quantized_variable.py (hunk 1 above)
Comment (lgeiger) [reply larq/quantized_variable.py]: Cool, thanks for checking :+1:

**Pull request merged**